src/dash-playlist-loader.js

```diff
@@ -409,35 +429,114 @@ export default class DashPlaylistLoader extends EventTarget {
    */
   refreshMedia_() {
     let oldMaster;
-    let newMaster;
+
+    const updateWithMaster = (newMaster) => {
+      const updatedMaster = updateMaster(oldMaster, newMaster);
+
+      if (updatedMaster) {
+        if (this.masterPlaylistLoader_) {
+          this.masterPlaylistLoader_.master = updatedMaster;
+        } else {
+          this.master = updatedMaster;
+        }
+        this.media_ = updatedMaster.playlists[this.media_.uri];
+      } else {
+        this.trigger('playlistunchanged');
+      }
+
+      if (!this.media().endList) {
+        this.mediaUpdateTimeout = window.setTimeout(() => {
+          this.trigger('mediaupdatetimeout');
+        }, refreshDelay(this.media(), !!updatedMaster));
+      }
+
+      this.trigger('loadedplaylist');
+    };
 
     if (this.masterPlaylistLoader_) {
       oldMaster = this.masterPlaylistLoader_.master;
-      newMaster = this.masterPlaylistLoader_.parseMasterXml();
+      this.masterPlaylistLoader_.parseMasterXml(updateWithMaster);
     } else {
       oldMaster = this.master;
-      newMaster = this.parseMasterXml();
+      this.parseMasterXml(updateWithMaster);
     }
+  }
 
-    const updatedMaster = updateMaster(oldMaster, newMaster);
+  addSidxInfoToPlaylists_(activeXhrs, master, doneFn) {
```

Comment:
Any reason to put the sidx helper functions (`addSidxInfoToPlaylists_`, `handleSidxResponse_`, `requestSidx_`, `fetchMediaSegmentsFromSidx_` ) onto the `DashPlaylistLoader` instance? None of them really require a reference to `this` (the only thing is a reference to xhr, which could easily be passed in as a param) and could instead be made into just functions that aren't attached to the class

Reply:
Ah, yeah that's a good point, most of those are just utilities 👍